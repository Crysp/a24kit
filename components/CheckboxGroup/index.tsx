import React from 'react';
import Checkbox from '../Checkbox';
import {
    List,
    Option,
} from './styled';

export interface Props {
    selected: string[]
    options: {
        value: any
        label: string
    }[]
    color: 'blue' | 'green' | 'deepBlue'
    large: boolean
    disabled: boolean
    onCheck(selected: string[]): void
}

interface State {
    selected: string[]
}

export default class CheckboxGroup extends React.PureComponent<Props, State> {
    static defaultProps = {
        selected: [],
        color: 'blue',
        large: false,
        disabled: false,
        onCheck() {},
    };

    state = {
        selected: this.props.selected || [],
    };

    onCheck = (value) => {
        const { selected } = this.state;
        const { onCheck } = this.props;
        const index = selected.indexOf(value);
        let nextSelected: string[] = [];
        if (index >= 0) {
            nextSelected = [
                ...selected.slice(0, index),
                ...selected.slice(index + 1, selected.length),
            ];
        } else {
            nextSelected = [...selected, value];
        }
        this.setState({ selected: nextSelected });
        onCheck(nextSelected);
    };

    render() {
        const {
            selected,
            options,
            color,
            large,
            disabled,
        } = this.props;
        return (
            <List>
                {options.map(({ value, label }) => (
                    <Option key={value}>
                        <Checkbox
                            label={label}
                            isChecked={selected.indexOf(value) >= 0}
                            onCheck={() => this.onCheck(value)}
                            color={color}
                            large={large}
                            disabled={disabled}
                        />
                    </Option>
                ))}
            </List>
        );
    }
}
