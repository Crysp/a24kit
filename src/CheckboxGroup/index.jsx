import React from 'react';
import PropTypes from 'prop-types';
import Checkbox from '../Checkbox';
import {
    List,
    Option,
} from './styled';


/**
 * @develop
 */
export default class CheckboxGroup extends React.PureComponent {
    static propTypes = {
        /**
         * Выбранные элементы
         */
        selected: PropTypes.arrayOf(PropTypes.string),
        /**
         * Набор данных для построения списка
         */
        options: PropTypes.arrayOf(PropTypes.shape({
            value: PropTypes.any,
            label: PropTypes.string,
        })).isRequired,
        /**
         * Цвет чекбоксов
         */
        color: PropTypes.oneOf(['blue', 'green', 'deepBlue']),
        /**
         * Большие чекбоксы
         */
        large: PropTypes.bool,
        /**
         * Неактивный список
         */
        disabled: PropTypes.bool,
        /**
         * Обработчик выбора элемента
         * @param value - выбранный элемент
         */
        onCheck: PropTypes.func,
    };

    static defaultProps = {
        selected: [],
        color: 'blue',
        large: false,
        disabled: false,
        onCheck() {},
    };

    state = {
        selected: this.props.selected || [],
    };

    onCheck = (value) => {
        const { selected } = this.state;
        const { onCheck } = this.props;
        const index = selected.indexOf(value);
        let nextSelected = [];
        if (index >= 0) {
            nextSelected = [
                ...selected.slice(0, index),
                ...selected.slice(index + 1, selected.length),
            ];
        } else {
            nextSelected = [...selected, value];
        }
        this.setState({ selected: nextSelected });
        onCheck(nextSelected);
    };

    render() {
        const {
            selected,
            options,
            color,
            large,
            disabled,
        } = this.props;
        return (
            <List>
                {options.map(({ value, label }) => (
                    <Option key={value}>
                        <Checkbox
                            label={label}
                            isChecked={selected.indexOf(value) >= 0}
                            onCheck={() => this.onCheck(value)}
                            color={color}
                            large={large}
                            disabled={disabled}
                        />
                    </Option>
                ))}
            </List>
        );
    }
}
