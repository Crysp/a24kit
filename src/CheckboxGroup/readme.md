```js
initialState = {
    selected: ["1","2","4"]
};
<CheckboxGroup
    selected={state.selected}
    options={[
        { value: "1", label: "Пункт 1" },
        { value: "2", label: "Пункт 2" },
        { value: "3", label: "Пункт 3" },
        { value: "4", label: "Пункт 4" },
        { value: "5", label: "Пункт 5" }
    ]}
    onCheck={selected => setState({ selected })}
/>
```