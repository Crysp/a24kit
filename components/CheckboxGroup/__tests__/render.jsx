import React from 'react';
import renderer from 'react-test-renderer';
import CheckboxGroup from '../index';


const optionsMock = [
    {
        value: '1',
        label: 'Label 1',
    },
    {
        value: '2',
        label: 'Label 2',
    },
];


describe('Верстка компонента <CheckboxGroup>', () => {
    it('Дефолтная', () => {
        const tree = renderer.create(<CheckboxGroup options={optionsMock} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Синие чекбоксы', () => {
        const tree = renderer.create(<CheckboxGroup options={optionsMock} color="blue" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Зеленые чекбоксы', () => {
        const tree = renderer.create(
            <CheckboxGroup options={optionsMock} color="green" />,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Темносиние чекбоксы', () => {
        const tree = renderer.create(
            <CheckboxGroup options={optionsMock} color="deepBlue" />,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Увеличенные чекбоксы', () => {
        const tree = renderer.create(<CheckboxGroup options={optionsMock} large />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Отмечаются выбранные', () => {
        const tree = renderer.create(
            <CheckboxGroup options={optionsMock} selected={['1']} />,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Неактивные чекбоксы', () => {
        const tree = renderer.create(<CheckboxGroup options={optionsMock} disabled />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
