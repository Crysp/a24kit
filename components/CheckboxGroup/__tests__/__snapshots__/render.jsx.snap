// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Верстка компонента <CheckboxGroup> Дефолтная 1`] = `
<div
  className="sc-bZQynM klYePH"
>
  <div
    className="sc-gzVnrw loObfe"
  >
    <div
      className="sc-ifAKCX gkzOrl"
      color="blue"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="sc-bwzfXH hwFftZ"
        disabled={false}
      >
        <div
          className="sc-htpNat jAdlEj"
          color="blue"
          disabled={false}
        >
          <svg
            aria-hidden="true"
            className="sc-bxivhb keOSFw sc-bdVaJa dTKkTG"
            focusable="false"
            size="auto"
            style={Object {}}
            viewBox="0 0 8 7"
          >
            <path
              d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
            />
          </svg>
        </div>
      </div>
      <div
        className="sc-EHOje hijERh"
        disabled={false}
      >
        Label 1
      </div>
    </div>
  </div>
  <div
    className="sc-gzVnrw loObfe"
  >
    <div
      className="sc-ifAKCX gkzOrl"
      color="blue"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="sc-bwzfXH hwFftZ"
        disabled={false}
      >
        <div
          className="sc-htpNat jAdlEj"
          color="blue"
          disabled={false}
        >
          <svg
            aria-hidden="true"
            className="sc-bxivhb keOSFw sc-bdVaJa dTKkTG"
            focusable="false"
            size="auto"
            style={Object {}}
            viewBox="0 0 8 7"
          >
            <path
              d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
            />
          </svg>
        </div>
      </div>
      <div
        className="sc-EHOje hijERh"
        disabled={false}
      >
        Label 2
      </div>
    </div>
  </div>
</div>
`;

exports[`Верстка компонента <CheckboxGroup> Зеленые чекбоксы 1`] = `
<div
  className="sc-bZQynM klYePH"
>
  <div
    className="sc-gzVnrw loObfe"
  >
    <div
      className="sc-ifAKCX ipiSZX"
      color="green"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="sc-bwzfXH hwFftZ"
        disabled={false}
      >
        <div
          className="sc-htpNat PhFqh"
          color="green"
          disabled={false}
        >
          <svg
            aria-hidden="true"
            className="sc-bxivhb keOSFw sc-bdVaJa dTKkTG"
            focusable="false"
            size="auto"
            style={Object {}}
            viewBox="0 0 8 7"
          >
            <path
              d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
            />
          </svg>
        </div>
      </div>
      <div
        className="sc-EHOje hijERh"
        disabled={false}
      >
        Label 1
      </div>
    </div>
  </div>
  <div
    className="sc-gzVnrw loObfe"
  >
    <div
      className="sc-ifAKCX ipiSZX"
      color="green"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="sc-bwzfXH hwFftZ"
        disabled={false}
      >
        <div
          className="sc-htpNat PhFqh"
          color="green"
          disabled={false}
        >
          <svg
            aria-hidden="true"
            className="sc-bxivhb keOSFw sc-bdVaJa dTKkTG"
            focusable="false"
            size="auto"
            style={Object {}}
            viewBox="0 0 8 7"
          >
            <path
              d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
            />
          </svg>
        </div>
      </div>
      <div
        className="sc-EHOje hijERh"
        disabled={false}
      >
        Label 2
      </div>
    </div>
  </div>
</div>
`;

exports[`Верстка компонента <CheckboxGroup> Неактивные чекбоксы 1`] = `
<div
  className="sc-bZQynM klYePH"
>
  <div
    className="sc-gzVnrw loObfe"
  >
    <div
      className="sc-ifAKCX epkeYp"
      color="blue"
      disabled={true}
      onClick={[Function]}
    >
      <div
        className="sc-bwzfXH jwtyQG"
        disabled={true}
      >
        <div
          className="sc-htpNat kjsDPm"
          color="blue"
          disabled={true}
        >
          <svg
            aria-hidden="true"
            className="sc-bxivhb keOSFw sc-bdVaJa dTKkTG"
            focusable="false"
            size="auto"
            style={Object {}}
            viewBox="0 0 8 7"
          >
            <path
              d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
            />
          </svg>
        </div>
      </div>
      <div
        className="sc-EHOje iBlIKd"
        disabled={true}
      >
        Label 1
      </div>
    </div>
  </div>
  <div
    className="sc-gzVnrw loObfe"
  >
    <div
      className="sc-ifAKCX epkeYp"
      color="blue"
      disabled={true}
      onClick={[Function]}
    >
      <div
        className="sc-bwzfXH jwtyQG"
        disabled={true}
      >
        <div
          className="sc-htpNat kjsDPm"
          color="blue"
          disabled={true}
        >
          <svg
            aria-hidden="true"
            className="sc-bxivhb keOSFw sc-bdVaJa dTKkTG"
            focusable="false"
            size="auto"
            style={Object {}}
            viewBox="0 0 8 7"
          >
            <path
              d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
            />
          </svg>
        </div>
      </div>
      <div
        className="sc-EHOje iBlIKd"
        disabled={true}
      >
        Label 2
      </div>
    </div>
  </div>
</div>
`;

exports[`Верстка компонента <CheckboxGroup> Отмечаются выбранные 1`] = `
<div
  className="sc-bZQynM klYePH"
>
  <div
    className="sc-gzVnrw loObfe"
  >
    <div
      className="sc-ifAKCX gkzOrl"
      color="blue"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="sc-bwzfXH jEjhxV"
        disabled={false}
      >
        <div
          className="sc-htpNat gqgabp"
          color="blue"
          disabled={false}
        >
          <svg
            aria-hidden="true"
            className="sc-bxivhb keOSFw sc-bdVaJa dTKkTG"
            focusable="false"
            size="auto"
            style={Object {}}
            viewBox="0 0 8 7"
          >
            <path
              d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
            />
          </svg>
        </div>
      </div>
      <div
        className="sc-EHOje hijERh"
        disabled={false}
      >
        Label 1
      </div>
    </div>
  </div>
  <div
    className="sc-gzVnrw loObfe"
  >
    <div
      className="sc-ifAKCX gkzOrl"
      color="blue"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="sc-bwzfXH hwFftZ"
        disabled={false}
      >
        <div
          className="sc-htpNat jAdlEj"
          color="blue"
          disabled={false}
        >
          <svg
            aria-hidden="true"
            className="sc-bxivhb keOSFw sc-bdVaJa dTKkTG"
            focusable="false"
            size="auto"
            style={Object {}}
            viewBox="0 0 8 7"
          >
            <path
              d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
            />
          </svg>
        </div>
      </div>
      <div
        className="sc-EHOje hijERh"
        disabled={false}
      >
        Label 2
      </div>
    </div>
  </div>
</div>
`;

exports[`Верстка компонента <CheckboxGroup> Синие чекбоксы 1`] = `
<div
  className="sc-bZQynM klYePH"
>
  <div
    className="sc-gzVnrw loObfe"
  >
    <div
      className="sc-ifAKCX gkzOrl"
      color="blue"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="sc-bwzfXH hwFftZ"
        disabled={false}
      >
        <div
          className="sc-htpNat jAdlEj"
          color="blue"
          disabled={false}
        >
          <svg
            aria-hidden="true"
            className="sc-bxivhb keOSFw sc-bdVaJa dTKkTG"
            focusable="false"
            size="auto"
            style={Object {}}
            viewBox="0 0 8 7"
          >
            <path
              d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
            />
          </svg>
        </div>
      </div>
      <div
        className="sc-EHOje hijERh"
        disabled={false}
      >
        Label 1
      </div>
    </div>
  </div>
  <div
    className="sc-gzVnrw loObfe"
  >
    <div
      className="sc-ifAKCX gkzOrl"
      color="blue"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="sc-bwzfXH hwFftZ"
        disabled={false}
      >
        <div
          className="sc-htpNat jAdlEj"
          color="blue"
          disabled={false}
        >
          <svg
            aria-hidden="true"
            className="sc-bxivhb keOSFw sc-bdVaJa dTKkTG"
            focusable="false"
            size="auto"
            style={Object {}}
            viewBox="0 0 8 7"
          >
            <path
              d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
            />
          </svg>
        </div>
      </div>
      <div
        className="sc-EHOje hijERh"
        disabled={false}
      >
        Label 2
      </div>
    </div>
  </div>
</div>
`;

exports[`Верстка компонента <CheckboxGroup> Темносиние чекбоксы 1`] = `
<div
  className="sc-bZQynM klYePH"
>
  <div
    className="sc-gzVnrw loObfe"
  >
    <div
      className="sc-ifAKCX kxdDuv"
      color="deepBlue"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="sc-bwzfXH hwFftZ"
        disabled={false}
      >
        <div
          className="sc-htpNat fWwtbt"
          color="deepBlue"
          disabled={false}
        >
          <svg
            aria-hidden="true"
            className="sc-bxivhb keOSFw sc-bdVaJa dTKkTG"
            focusable="false"
            size="auto"
            style={Object {}}
            viewBox="0 0 8 7"
          >
            <path
              d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
            />
          </svg>
        </div>
      </div>
      <div
        className="sc-EHOje hijERh"
        disabled={false}
      >
        Label 1
      </div>
    </div>
  </div>
  <div
    className="sc-gzVnrw loObfe"
  >
    <div
      className="sc-ifAKCX kxdDuv"
      color="deepBlue"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="sc-bwzfXH hwFftZ"
        disabled={false}
      >
        <div
          className="sc-htpNat fWwtbt"
          color="deepBlue"
          disabled={false}
        >
          <svg
            aria-hidden="true"
            className="sc-bxivhb keOSFw sc-bdVaJa dTKkTG"
            focusable="false"
            size="auto"
            style={Object {}}
            viewBox="0 0 8 7"
          >
            <path
              d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
            />
          </svg>
        </div>
      </div>
      <div
        className="sc-EHOje hijERh"
        disabled={false}
      >
        Label 2
      </div>
    </div>
  </div>
</div>
`;

exports[`Верстка компонента <CheckboxGroup> Увеличенные чекбоксы 1`] = `
<div
  className="sc-bZQynM klYePH"
>
  <div
    className="sc-gzVnrw loObfe"
  >
    <div
      className="sc-ifAKCX gkzOrl"
      color="blue"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="sc-bwzfXH cdMfyU"
        disabled={false}
      >
        <div
          className="sc-htpNat jAdlEj"
          color="blue"
          disabled={false}
        >
          <svg
            aria-hidden="true"
            className="sc-bxivhb jIZPyy sc-bdVaJa dTKkTG"
            focusable="false"
            size="auto"
            style={Object {}}
            viewBox="0 0 8 7"
          >
            <path
              d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
            />
          </svg>
        </div>
      </div>
      <div
        className="sc-EHOje hijERh"
        disabled={false}
      >
        Label 1
      </div>
    </div>
  </div>
  <div
    className="sc-gzVnrw loObfe"
  >
    <div
      className="sc-ifAKCX gkzOrl"
      color="blue"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="sc-bwzfXH cdMfyU"
        disabled={false}
      >
        <div
          className="sc-htpNat jAdlEj"
          color="blue"
          disabled={false}
        >
          <svg
            aria-hidden="true"
            className="sc-bxivhb jIZPyy sc-bdVaJa dTKkTG"
            focusable="false"
            size="auto"
            style={Object {}}
            viewBox="0 0 8 7"
          >
            <path
              d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
            />
          </svg>
        </div>
      </div>
      <div
        className="sc-EHOje hijERh"
        disabled={false}
      >
        Label 2
      </div>
    </div>
  </div>
</div>
`;
